src/smtsolvers/CoreSMTSolver.cc

```diff
@@ -2251,9 +2252,42 @@ void CoreSMTSolver::fillBooleanVars(ModelBuilder &modelBuilder) {
     Logic& logic = theory_handler.getLogic();
     for (Var v = 0; v < model.size(); ++v) {
         PTRef atom = theory_handler.varToTerm(v);
-        if (logic.isBoolAtom(atom) && model[v] != l_Undef) {
-            PTRef val = model[v] == l_True ? logic.getTerm_true() : logic.getTerm_false();
+        PTRef target_tr = theory_handler.getSubstitution(atom);
```

Comment:
Here I was also hoping we can get rid of the substitutions when building the model...

Reply:
I have also some experimental evidence now that this code is not necessary.